classes/DepositObjectDAO.inc.php

```diff
@@ -27,21 +27,17 @@ public function getById($journalId, $depositObjectId) {
 			[(int) $journalId, (int) $depositObjectId]
 		);
 
-		$returner = null;
-		if ($result->RecordCount() != 0) {
-			$returner = $this->_fromRow($result->GetRowAssoc(false));
-		}
-
-		$result->Close();
+		$row = $result->current();
```

Comment:
For cases like this I've been using ternary syntax:
```
$row = $result->current();
return $row ? $this->_fromRow((array) $row) : null;
```
It's a little clearer (IMO) what the return type will be -- otherwise it looks like it can return either `DepositObject` (from `_fromRow`) or whatever `$row` is.